plaso/parsers/locate.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+"""Parser for locate/updatedb database files."""
+
+import os
+
+from struct import unpack as struct_unpack
+from struct import error as struct_error
+
+from dfdatetime import posix_time
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import definitions
+from plaso.lib import errors
+from plaso.lib import specification
+from plaso.parsers import interface
+from plaso.parsers import manager
+
+
+class LocateDatabaseEvent(events.EventData):
+  """Linux Locate Database event data.
+
+  Attributes:
+    folder_path (str): full folder path.
+  """
+
+  DATA_TYPE = 'linux:locate'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(LocateDatabaseEvent, self).__init__(data_type=self.DATA_TYPE)
+    self.folder_path = None
+
+
+class LocateDatabaseFile(object):
+  """Locate database file."""
+
+  DB_MAGIC = b"\x00mlocate"
+
+  # header struct format
+  _DB_HEADER = ">8sIbb2s"
+
+  # directory struct format
+  _DB_DIRECTORY = ">QI4s"
+
+  # Entry Types
+  _DBE_NORMAL = 0
+  _DBE_DIRECTORY = 1
+  _DBE_END = 2
+
+  def __init__(self):
+    """Initialises a locate database file."""
+    super(LocateDatabaseFile, self).__init__()
+    self.root_path = None
```

Comment:
per style guide: if this is a public attribute it needs to be defines in LocateDatabaseFile docstring.

Reply:
Updated docstring to include public attribute